src/java.base/share/classes/java/lang/Module.java

```diff
@@ -282,53 +290,70 @@ private static boolean trySetEnableNativeAccess(Module target) {
-            if (illegalNativeAccess == ModuleBootstrap.IllegalNativeAccess.DENY) {
-                throw new IllegalCallerException("Illegal native access from " + mod);
-            } else if (EnableNativeAccess.trySetEnableNativeAccess(target)) {
-                // warn and set flag, so that only one warning is reported per module
-                String cls = owner.getName();
-                String mtd = cls + "::" + methodName;
-                String modflag = isNamed() ? getName() : "ALL-UNNAMED";
-                String caller = currentClass != null ? currentClass.getName() : "code";
-                if (jni) {
-                    VM.initialErr().printf("""
+            ensureNativeAccessSlowPath(owner, methodName, currentClass, jni, target, illegalNativeAccess);
+        }
+    }
+
+    @DontInline
+    void ensureNativeAccessSlowPath(Class<?> owner,
+                                    String methodName,
+                                    Class<?> currentClass,
+                                    boolean jni,
+                                    Module target,
+                                    ModuleBootstrap.IllegalNativeAccess illegalNativeAccess) {
+        String mod = mod(currentClass);
+        if (illegalNativeAccess == ModuleBootstrap.IllegalNativeAccess.DENY) {
+            throw new IllegalCallerException("Illegal native access from " + mod);
+        } else if (EnableNativeAccess.trySetEnableNativeAccess(target)) {
+            // warn and set flag, so that only one warning is reported per module
+            String cls = owner.getName();
+            String mtd = cls + "::" + methodName;
+            String modflag = isNamed() ? getName() : "ALL-UNNAMED";
+            String caller = currentClass != null ? currentClass.getName() : "code";
+            if (jni) {
+                VM.initialErr().printf("""
                             WARNING: A native method in %s has been bound
                             WARNING: %s is declared in %s
                             WARNING: Use --enable-native-access=%s to avoid a warning for native methods declared in this module
                             WARNING: Restricted methods will be blocked in a future release unless native access is enabled
                             %n""", cls, mtd, mod, modflag);
-                } else {
-                    VM.initialErr().printf("""
+            } else {
+                VM.initialErr().printf("""
                             WARNING: A restricted method in %s has been called
                             WARNING: %s has been called by %s in %s
                             WARNING: Use --enable-native-access=%s to avoid a warning for callers in this module
                             WARNING: Restricted methods will be blocked in a future release unless native access is enabled
                             %n""", cls, mtd, caller, mod, modflag);
+            }
+        }
+    }
+
+    private String mod(Class<?> currentClass) {
```

Comment:
Can you try to find a better name for this method as mod will be read as "modifiers" or "module name" and it's either.